eatin-client/src/components/functions/SavedStateInRecipes.ts

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useEffect, useState } from "react";
+import { useAuth } from "../../context/auth-context";
+import { useGetUserrecipesByUserIdQuery } from "../../generated/graphql";
+import { Recipe, RecipesSection } from "../types";
+
+export function useAddIsSavedToRecipesSection(givenRecipes: RecipesSection[]) {
+    const { currentUser } = useAuth();
+    const { data: userRecipesData } = useGetUserrecipesByUserIdQuery({
+        variables: { userID: currentUser ? currentUser.uid : "" },
+    });
+    const [updatedRecipes, setNewRecipes] = useState<RecipesSection[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const addIsSavedToRecipesSection = useCallback(() => {
+        setIsLoading(true);
+        const updatedStateRecipes = givenRecipes.map((section: RecipesSection) => {
+            return {
+                ...section,
+                recipes: section.recipes.map((recipe: Recipe) => {
+                    let isRecipeSaved = false;
+                    if (userRecipesData) {
+                        const index = userRecipesData.userRecipesByUser.findIndex(
```

Comment:
you can use `find` instead of `findIndex`, which returns the first occurrence of the predicate

Reply:
but then it would not let me change the object in the array, no?